lib/shelley/src/Cardano/Wallet/Shelley/Network.hs

```diff
@@ -329,8 +341,36 @@ withNetworkLayer tr np addrInfo versionData action = do
             SubmitSuccess -> pure ()
             SubmitFail err -> throwE $ ErrPostTxBadRequest $ T.pack (show err)
 
-    _stakeDistribution =
-        notImplemented "stakeDistribution"
+    handleQueryFailure = withExceptT
+        (\e -> ErrNetworkUnreachable $ T.pack $ "Unexpected" ++ show e) . ExceptT
+    _stakeDistribution queue pt coin = do
+        stakeMap <- fromPoolDistr <$> handleQueryFailure
+            (queue `send` CmdQueryLocalState pt OC.GetStakeDistribution)
+        let toStake = Set.singleton $ Left $ toShelleyCoin coin
+        rewardsPerAccount <- fromNonMyopicMemberRewards <$> handleQueryFailure
+            (queue `send` CmdQueryLocalState pt (OC.GetNonMyopicMemberRewards toStake))
+        pparams <- handleQueryFailure
+            (queue `send` CmdQueryLocalState pt OC.GetCurrentPParams)
```

Comment:
This one feels quite redundant / awkward. It sure is needed by the caller to compute few things on the pool data, but the caller should simply call `getProtocolParameters`, don't you think ?

Reply:
This code guarantees that the `pparams` are valid at the same point `pt`.

If we were to instead use
```haskell
getProtocolParameters :: m ProtocolParameters
```
We lose that control, and introduce potential race conditions. Even if unlikely, _that_ feels incredibly wrong to me.